File: python/quicksort.py
import random
import timeit

arr = [0] * 100000

for i in range(len(arr)):
  arr[i] = random.randrange(1,1000001)


def partition(arr,low,high): 
    i = ( low-1 )         
    pivot = arr[high]     
  
    for j in range(low , high): 
  
        if   arr[j] < pivot: 
          
            i = i+1 
            arr[i],arr[j] = arr[j],arr[i] 
  
    arr[i+1],arr[high] = arr[high],arr[i+1] 
    return ( i+1 ) 

def quickSort(arr,low,high): 
    if low < high: 
  
        pi = partition(arr,low,high) 
  
        quickSort(arr, low, pi-1) 
        quickSort(arr, pi+1, high) 
  

start = timeit.default_timer()
quickSort(arr,0, len(arr)-1)
stop = timeit.default_timer()
print("Quicksort Time", stop - start)

File: java/QuickSort.java
package homework200323;

import java.util.Random;

public class QuickSort {
	
	static int partition(int[] arr,int l,int h) {
		
		int pivot = arr[h];
		int i = (l-1);
		for(int j = l; j < h; j++) {
			if(arr[j] <= pivot) {
				i++;
				int temp = arr[i];
				arr[i] = arr[j];
				arr[j] = temp;
			}
		}
		int temp = arr[i+1];
		arr[i+1] = arr[h];
		arr[h] = temp;
		return i+1;
	}
	
	static void sort(int[] arr, int l, int h) {
		if(l < h) {
			int p = partition(arr,l,h);
			sort(arr,l,p-1);
			sort(arr,p+1,h);
		}
	}
	
	public static void main(String[] args) {
		int[] arr = new int[1000000];
		Random r = new Random();
		for(int i = 0; i < arr.length; i++) {
			arr[i] = r.nextInt(1000000);
		}
		long start = System.currentTimeMillis();
		sort(arr,0,arr.length-1);
		long stop = System.currentTimeMillis();
		System.out.println("Quicksort Time: " + (stop-start));
	}
}
